packages/marshaller/src/ddl2/dashboard.ts

```diff
@@ -331,20 +354,32 @@ export class Dashboard extends ChartPanel {
         });
     }
 
-    constructor(private _ec: ElementContainer) {
+    constructor(protected _ec: ElementContainer) {
         super();
         this._ec.on("vizStateChanged", (viz) => {
             this.vizStateChanged(viz);
         });
+        this.initializeLayoutWidget();
+        this
+            .buttons([
+                this._designModeButton,
+                this._addButton,
+                this._removeButton,
+                new Spacer(),
+                this._reloadButton,
+                new Spacer(),
+                this._addSamples
+            ])
+            .widget((this._dockPanel as unknown) as Widget)
+            ;
+    }
+
+    initializeLayoutWidget() {
         this._dockPanel = new DashboardDockPanel(this._ec)
             .on("vizActivation", (elem: Element) => {
                 this.vizActivation(elem);
             })
             ;
-        this
-            .buttons([this._addButton, this._removeButton, new Spacer(), this._reloadButton, new Spacer(), this._addSamples])
-            .widget(this._dockPanel)
-            ;
```

Comment:
Was there a reason this code got moved?

Reply:
Originally it was to allow `Dashboard2` to simply overwrite that `initializeLayoutWidget` method in order to use `DashboardGrid` instead of `DashboardDockPanel`, but that's no longer necessary since the distinction between those two dashboard classes happens in the dashy restore method. I'll revert this part.